pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+                the CIM namespace, or `None` if the server did not return host
+                information.
+
+            * If  ObjectName is :term:`string` (class-level): A list of
+              :class:`~pywbem.CIMClassName` objects that are the class paths of
+              the associated classes, with their attributes set as follows:
+
+              * `classname`: Name of the class.
+              * `namespace`: Name of the CIM namespace containing the class.
+              * `host`: Host and optionally port of the WBEM server containing
+                the CIM namespace, or `None` if the server did not return host
+                information.
+
+        Raises:
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_PARAMETER
+        """
+
+        if isinstance(ObjectName, CIMInstanceName):
+            rtn_paths = self._get_associated_instancenames(namespace,
+                                                           ObjectName,
+                                                           AssocClass,
+                                                           ResultClass,
+                                                           ResultRole, Role)
+            results = [p.copy() for p in rtn_paths]
+
+            for iname in results:
+                if iname.host is None:
+                    iname.host = self.host
+
+        else:
+            rtn_classnames = self._get_associated_classnames(namespace,
+                                                             ObjectName,
+                                                             AssocClass,
+                                                             ResultClass,
+                                                             ResultRole, Role)
+            # TODO: Future Returns CIMClassName. Should be just the classname
+            # from a server since CIMClassName is a client artifact and the
+            # server has no way to send that to client.
+            results = [CIMClassName(classname=cln, host=self.host,
+                                    namespace=namespace)
+                       for cln in rtn_classnames]
+
+        return results
+
+    ####################################################################
+
+    #   Associators server request
+    #
+    ####################################################################
+
+    def Associators(self, namespace, ObjectName, AssocClass=None,
+                    ResultClass=None,
+                    Role=None, ResultRole=None, IncludeQualifiers=None,
+                    IncludeClassOrigin=None, PropertyList=None):
+        """
+        Implements a WBEM server responder for
+            :meth:`~pywbem.WBEMConnection.Associators`
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
DONE